nft-client/src/main.rs

```diff
@@ -0,0 +1,635 @@
+            println!("Setting up PostgreSQL trigger and notifications.");
+            db_client
+                .execute(
+                    format!(
+                        "CREATE FUNCTION pg_temp.notify_insert () RETURNS TRIGGER AS $psql$
+                        BEGIN
+                            IF NEW.index = {} AND NEW.subindex = {} THEN
+                                PERFORM pg_notify('contract_updates', NEW.summary::text);
+                            END IF;
+                            RETURN NEW;
+                        END;$psql$ LANGUAGE plpgsql",
+                        contract.0.index.index, contract.0.subindex.sub_index
+                    )
+                    .as_str(),
+                    &[],
+                )
+                .await
+                .context("Failed creating PostgreSQL 'notify_insert' function")?;
+
+            db_client
+                .batch_execute(
+                    "CREATE TRIGGER notify_on_insert_into_cti AFTER INSERT ON cti FOR EACH ROW \
+                     EXECUTE PROCEDURE pg_temp.notify_insert();
+                         LISTEN contract_updates;",
+                )
+                .await
+                .context("Failed creating PostgreSQL trigger")?;
+
+            // Await notifications, find and parse the relevant transaction summaries
+            println!("Started tracing contract events:");
+            while let Some(notification) = notifications.next().await {
+                if notification.channel() != "contract_updates" {
+                    continue;
+                }
+                let summary_index: i64 = notification.payload().parse()?;
+
+                let row = db_client
+                    .query_one("SELECT summary FROM summaries WHERE id = $1", &[
+                        &summary_index,
+                    ])
+                    .await?;
+                let summary = serde_json::from_value::<DatabaseSummaryEntry>(row.get(0))?;
+                let events = collect_summary_contract_events(summary, contract.0);
+                for event in events {
+                    let mut cursor = concordium_contracts_common::Cursor::new(event.as_ref());
+                    let event = cts1::Event::deserial(&mut cursor)
+                        .map_err(|_| anyhow!("Failed parsing event"))?;
+                    println!("{}", event);
+                }
+            }
+        }
+        Command::Mint {
+            sender,
+            token_ids,
+            contract,
+        } => {
+            // The GRPC client for Concordium node.
+            let mut grpc_client = Client::connect(app.endpoint, "rpcadmin".to_string())
+                .await
+                .context("Failed to connect to Node GRPC")?;
+
+            let account_data: AccountData = serde_json::from_str(
+                &std::fs::read_to_string(sender).context("Could not read the keys file.")?,
+            )
+            .context("Could not parse the accounts file.")?;
```

Comment:
Recommend moving this before the establish connection call. No need to establish a connection if this fails.